StandaloneMmPkg/Drivers/StandaloneMmCpu/EventHandle.c

```diff
@@ -72,77 +67,68 @@ STATIC EFI_MM_ENTRY_POINT  mMmEntryPoint = NULL;
+  if (CommBufferAddr == (UINTN)NULL) {
     return EFI_INVALID_PARAMETER;
   }
 
+  if (mMmEntryPoint == NULL) {
+    DEBUG ((DEBUG_ERROR, "Mm Entry point Not Found\n"));
+    return EFI_UNSUPPORTED;
+  }
+
   // Find out the size of the buffer passed
-  NsCommBufferSize = ((EFI_MM_COMMUNICATE_HEADER *)NsCommBufferAddr)->MessageLength +
-                     sizeof (EFI_MM_COMMUNICATE_HEADER);
+  CommBufferSize = ((EFI_MM_COMMUNICATE_HEADER *)CommBufferAddr)->MessageLength +
+                   sizeof (EFI_MM_COMMUNICATE_HEADER);
 
-  GuidedEventContext = NULL;
   // Now that the secure world can see the normal world buffer, allocate
   // memory to copy the communication buffer to the secure world.
   Status = mMmst->MmAllocatePool (
                     EfiRuntimeServicesData,
-                    NsCommBufferSize,
-                    (VOID **)&GuidedEventContext
+                    CommBufferSize,
+                    (VOID **)&gGuidedEventContext
                     );
 
-  if (Status != EFI_SUCCESS) {
+  if (EFI_ERROR (Status)) {
+    gGuidedEventContext = NULL;
     DEBUG ((DEBUG_ERROR, "Mem alloc failed - 0x%x\n", EventId));
-    return EFI_OUT_OF_RESOURCES;
+    return Status;
   }
 
-  // X1 contains the VA of the normal world memory accessible from
-  // secure world.
-  CopyMem (GuidedEventContext, (CONST VOID *)NsCommBufferAddr, NsCommBufferSize);
-
-  // Stash the pointer to the allocated Event Context for this CPU
-  PerCpuGuidedEventContext[CpuNumber] = GuidedEventContext;
+  CopyMem (gGuidedEventContext, (CONST VOID *)CommBufferAddr, CommBufferSize);
 
   ZeroMem (&MmEntryPointContext, sizeof (EFI_MM_ENTRY_CONTEXT));
 
-  MmEntryPointContext.CurrentlyExecutingCpu = CpuNumber;
-  MmEntryPointContext.NumberOfCpus          = mMpInformationHobData->NumberOfProcessors;
+  // StandaloneMm UP-migratable which means it cannot run concurrently.
+  // Therefore, set number of cpus as 1 and cpu number as 0.
+  MmEntryPointContext.CurrentlyExecutingCpu = 0;
+  MmEntryPointContext.NumberOfCpus          = 1;
 
   // Populate the MM system table with MP and state information
-  mMmst->CurrentlyExecutingCpu = CpuNumber;
-  mMmst->NumberOfCpus          = mMpInformationHobData->NumberOfProcessors;
+  mMmst->CurrentlyExecutingCpu = 0;
+  mMmst->NumberOfCpus          = 1;
   mMmst->CpuSaveStateSize      = 0;
   mMmst->CpuSaveState          = NULL;
 
-  if (mMmEntryPoint == NULL) {
-    DEBUG ((DEBUG_ERROR, "Mm Entry point Not Found\n"));
-    return EFI_UNSUPPORTED;
-  }
-
   mMmEntryPoint (&MmEntryPointContext);
 
   // Free the memory allocation done earlier and reset the per-cpu context
-  ASSERT (GuidedEventContext);
-  CopyMem ((VOID *)NsCommBufferAddr, (CONST VOID *)GuidedEventContext, NsCommBufferSize);
-
-  Status = mMmst->MmFreePool ((VOID *)GuidedEventContext);
-  if (Status != EFI_SUCCESS) {
-    return EFI_OUT_OF_RESOURCES;
-  }
+  CopyMem ((VOID *)CommBufferAddr, (CONST VOID *)gGuidedEventContext, CommBufferSize);

```

Comment:
Please update the comment above.

Reply:
No need to update as above. It's a way to change.